### PR TITLE
IACOE-73 + checkout page

### PR DESCRIPTION
**Pull request by kirkate**

### CODE REVIEW

File: frontend/src/pages/Store/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+            history.replace('/store/iphone');
+        }
+    }, []);
+    const [activeSubIdCategory, setActiveSubIdCategory] = useState(null);
+    function handleSubCategoryClick(subCategoryId, event) {
+        event.preventDefault();
+        history.push(`${location.pathname}/${subCategoryId}`);
+        setActiveSubIdCategory(subCategoryId);
+    }
+    return (h(Container, null,
```
Comment (dvada) [inline frontend/src/pages/Store/index.js]: what is that?!

File: frontend/src/pages/Store/index.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+export const Store = () => {
+    const history = useHistory();
+    const location = useLocation();
+    useEffect(() => {
+        if (history.location.pathname === '/store') {
+            history.replace('/store/iphone');
+        }
+    }, []);
+    const [activeSubIdCategory, setActiveSubIdCategory] = useState(null);
+    function handleSubCategoryClick(subCategoryId, event) {
```
Comment (dvada) [inline frontend/src/pages/Store/index.js]: wrap into useCallback

File: frontend/src/pages/Checkout/index.tsx [hunk 3]
```diff
@@ -1,3 +1,50 @@
...
+} from '../../config/config';
+import { RadioButton } from '../../components/Form/RadioButton';
+import { ConditionalSection } from '../../components/Form/ConditionalSection';
+import { ProductsList } from '../../components/ProductsList';
 
-export const Checkout = () => <p>Checkout</p>;
+export const Checkout = () => (
+  <section class="checkoutPage">
+    <ProductsList />
+    <Form initialValues={initialValues} validations={validations} class="deliveryForm">
```
Comment (dvada) [inline frontend/src/pages/Checkout/index.tsx]: Move to CheckoutForm component

File: frontend/src/context/DataContext.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { h, createContext } from 'preact';
```
Comment (dvada) [inline frontend/src/context/DataContext.tsx]: does it use?

File: frontend/src/components/Form/hooks/useForm.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+      [name]: ValidationUtil.validateField(name, value, validations),
+    }));
+  }, []);
+
+  const handleSubmit = useCallback((event) => {
+    event.preventDefault();
+    const errorCheck = ValidationUtil.validate(values, validations);
+    setErrors(errorCheck);
+    setIsSubmitted(true);
+    if (isSubmitted) {
```
Comment (dvada) [inline frontend/src/components/Form/hooks/useForm.tsx]: remove this block

File: frontend/src/components/Form/hooks/useFields.tsx [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import { h } from 'preact';
```
Comment (dvada) [inline frontend/src/components/Form/hooks/useFields.tsx]: does it use?

File: frontend/src/components/Form/FieldsetRadio.tsx [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import { h } from 'preact';
+import { useFormContext } from './context';
+
+export const RadioGroup = ({ children, name }) => {
+  const {
+    errors, handleChange, values,
+  } = useFormContext();
+
+  return (
+    <fieldset onChange={handleChange} currentValue={values[name]}>
```
Comment (dvada) [inline frontend/src/components/Form/FieldsetRadio.tsx]: how it works?

File: frontend/src/pages/Checkout/index.tsx [hunk 8]
```diff
@@ -1,3 +1,50 @@
...
+        </Fieldset>
+        <Fieldset label="Post office №" name="postOffice">
+          <input name="postOffice" type="text" />
+        </Fieldset>
+      </ConditionalSection>
+      <ConditionalSection name="deliveryType" value={deliveryType.pickupInStore}>
+        <address>Address: Antonovycha St, 176, Kyiv, 03150</address>
+      </ConditionalSection>
+      <ConditionalSection name="deliveryType" value={deliveryType.сourier}>
+        <RadioGroup>
```
Comment (dvada) [inline frontend/src/pages/Checkout/index.tsx]: RadioGroup should have a name prop

File: frontend/src/components/Form/hooks/useForm.tsx (hunk 5 above)
Comment (kirkate) [reply frontend/src/components/Form/hooks/useForm.tsx:34]: If remove this block, the validation on the submission dosnt work.

**Pull request merged**